pkg/resourcemanager/azuresql/azuresqlshared/sqlproperties.go

```diff
@@ -78,48 +96,77 @@ func SQLServerPropertiesToServer(properties SQLServerProperties) (result sql.Ser
 func SQLDatabasePropertiesToDatabase(properties SQLDatabaseProperties) (result sql.DatabaseProperties) {
 
 	result = sql.DatabaseProperties{
-		Edition: translateDBEdition(properties.Edition),
+		MaxSizeBytes: properties.MaxSizeBytes,
+	}
+
+	return result
+}
+
+// SQLDatabasePropertiesToDatabase translates SQLDatabaseProperties to DatabaseProperties
+func SQLDatabasePropertiesToSku(properties SQLDatabaseProperties) (result *sql.Sku) {
+
+	if properties.Sku == nil {
+		return nil
+	}
+
+	result = &sql.Sku{
+		Name:     &properties.Sku.Name,
+		Tier:     &properties.Sku.Tier,
+		Size:     &properties.Sku.Size,
+		Family:   &properties.Sku.Family,
+		Capacity: properties.Sku.Capacity,
 	}
 
 	return result
 }
 
-// translateDBEdition translates enums
-func translateDBEdition(in v1alpha1.DBEdition) (result sql.DatabaseEdition) {
+func MergeDBEditionAndSku(in v1beta1.DBEdition, sku *v1beta1.SqlDatabaseSku) (*Sku, error) {
+
+	// There are 3 cases:
+	// - Customer just sets Spec.Sku - Spec.Sku is sent as is to ARM.
+	// - Customer just sets Spec.Edition - Edition is converted to Spec.Sku and that is sent to ARM
+	// - Customer sets both Spec.Sku and Spec.Edition - Spec.Edition is ignored and Spec.Sku is used
+
+	if sku != nil {
+		return &Sku{
+			Name:     sku.Name,
+			Tier:     sku.Tier,
+			Size:     sku.Size,
+			Family:   sku.Family,
+			Capacity: sku.Capacity,
+		}, nil
+	}
+
+	// These defaults were determined based on what SQL defaults to in the 2015-03-01-preview API.
+	// The way this works is that for editions that were DTU based, Edition maps to name: Edition, tier: Edition,
+	// so for example edition: "Basic" became name: "Basic", tier: "Basic". For VCore offerings, the name can't actually
+	// be derived from just the old edition, so we had to hardcode some actual values (the defaults which SQL happens
+	// to use - i.e. HS_Gen5_2 is the default when you deploy a 2015-03-01-preview SQL database with Edition: "Hyperscale").
+	// There is some risk that if SQL changes their defaults we'll be out of sync, but given we're planning to remove
+	// edition in a future API version and customers have full access to SKU to specify exactly what they want, that
+	// shouldn't really be a problem.
 	switch in {
 	case 0:
```

Comment:
can we use named constants here?

Reply:
Yes -- although the code actually didn't before. The customer just specified it as a number (which seemed weird to me -- not clear where they got the mapping between number and meaning...